### ISSUE TITLE
Anidb ID and other sites

### ISSUE THREAD

**Issue by seertenedos** [groenlid/xdm-anime-plugin-repo#10]
Is there any chance your API can return the AniDB ID and add it to the anime so that i can match stuff found on AniDB to your series?  Also been able to search by AniDB AID would be fantastic as well.

I think Lad wanted the other ID's as well to do plugins for other sites that can enhance the info or file processing.  I want to focus on file processing but i need AID and EID for AniDB to do that.  With just AID i should be able to work out the EID's with you don't have those.

I know it is not a defect but this is the only way i know how to contact you

**Comment by seertenedos** [comment]
sorry just noticed the data at the bottom of the API response.  I will use that.

**Comment by groenlid** [comment]
So, like you said. There should be a connection-array in the animedetails request which maps the anime to anidb, thetvdb, themoviedb and myanimelist. Hope you can use that :-) I've also added (but not deployed) a connection-array under the episodedetails route for episode-specific mapping, but I would not use this yet as only a very limited amount of episodes are mapped atm.

Hope this helps and let me know if you need anything else.

Seer notifications@github.com wrote:

> Is there any chance your API can return the AniDB ID and add it to the
> anime so that i can match stuff found on AniDB to your series?  Also
> been able to search by AniDB AID would be fantastic as well.
> 
> I think Lad wanted the other ID's as well to do plugins for other sites
> that can enhance the info or file processing.  I want to focus on file
> processing but i need AID and EID for AniDB to do that.  With just AID
> i should be able to work out the EID's with you don't have those.
> 
> I know it is not a defect but this is the only way i know how to
> contact you
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/groenlid/xdm-anime-plugin-repo/issues/10

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by seertenedos** [comment]
I think it is fine for now.  I just looked at Lad's latest code and it is integrated in his so i think i will be fine for now.

With the EID does it need to be another route or could it just be added under the episode info on the animedetails route? I have no issue updating your plugin to make the call for every episode but seems like it would put a lot of load on your server to just get a few extra numbers.  Also are you able to supply a few animes that have this info or could you update the gundam seed to have it as i have the files for that so i could use it for testing. Lastly is this info auto extracted or are you manually populating it? If auto extracting i assume you know already but the http api is the easiest way to get series info and should have every eid for a series.

**Comment by seertenedos** [comment]
hi groenlid,
Any update on the connection array for episodes? Any chance of putting them on the AnimeDetails service under each episode as well? If this is unlikely i will look at sicking the data from AniDb but since you seem to already have the data it seems better to use that if possible as it saves me from rate limiting and matching episodes between yours and AniDB.

**Comment by groenlid** [comment]
I'm currently restructuring the api and database to support the episode-connections, so it shouldn't be too long.
One thing I could do in the meantime is to make some test data for you, so you can start the development.

I'm still a little uncertain under which resource the connection should lie.

Either under under episodeDetails where you could get fetch one with the url /episodeDetails/1 and multiple with /episodeDetails/?ids[]=1&ids[]=2.

Or just add it under animeDetails.episodes.

I'll look into it later today :+1: to see what's best practice.

**Comment by seertenedos** [comment]
yea some test data would be good. At the min i would prefer to modify your plugin to suck in the data than hook into anidb for that due to the restrictions from anidb plus all i want is the eid and not the rest of the data. Even if you choose to put it under the episode details it would be great to have a api for getting the details that XDM needs with minimal hits to your service. 

Say i want to refresh my collection of 100 anime. If you have all the info needed on one service call it is 100 hits if not then it is 100 hit + 100x30(assuming average of 30 eps per series) so 3100 hits to your service where 3k of those hits are just to get a single number.  Also something else i noticed is the API currently returns the ratings and people that have watched it.  That could get huge very easily that would be better as an option you could add to the request if you want it or as a summary but not a big deal. 

Basically i am trying to use http://anidb.net/perl-bin/animedb.pl?show=anime&aid=9960 as a test anime with my target to get AniDB to recognize my existing files and return an EID that i match to the episode info you return and then use to mark it as downloaded and process the file. Later on i want it scan exiting folders and process them and also add any missing anime into the system but just getting it to process 1 file is my first goal.

**Comment by groenlid** [comment]
I see your point, which was also the reason that I also leaned towards the "add it under animeDetails route" solution.

The episodeDetails should in the future just include cached statistics and "last seen by" information associated to the episode, but you know how it is.. // TODO: Cache this and make it better.

I'll add the test-data after work today :+1:

**Comment by seertenedos** [comment]
Thanks so much! Great work so far and can't wait till I can build a plugin
to add to it. Can you past here when done so I know.  Thanks

**Comment by groenlid** [comment]
Should work now.
http://groenlid.no-ip.org/api/animedetails/1302

Though this was done through manual inserts, it will give you a feel for how the payload-structure will look like :+1:

**Comment by seertenedos** [comment]
perfect! thanks so much. Can you let me know when you get it happening for
all but in the mean time this is enough to build my plugin and do basic
testing!

**Comment by seertenedos** [comment]
I have been working with LAD on an anime file importer/processor that uses AniDB which is slowly getting there but i have hit an issue.  Any chance you could add an API call to get anime details but anidb aid as well? Just for the search?

**Comment by groenlid** [comment]
Not quite sure what you're asking @seertenedos.

>  Any chance you could add an API call to get anime details but anidb aid as well?
> Just for the search?

Do you mean, that you want search capability based on anidb-id (cause there is already a endpoint called animedetails with a connection-array to anidb)?

**Comment by seertenedos** [comment]
I need this for example
http://groenlid.no-ip.org/api/anime?aid=146767
Where aid is the AniDB anime ID.  Basically i need an API to your series ID based on the AID i get from AniDB.  Then i can import new series based on the files found in a folder. The idea is people can import an existing collection and it will auto add the shows to their collect using info from your site but the files are looked up on AniDB to get AID and EID so it knows exactly what each file is.

**Comment by groenlid** [comment]
Sure, we could do something like this. I'll look into it tomorrow :-)

**Comment by seertenedos** [comment]
Thanks

**Comment by seertenedos** [comment]
Also any chance you can load the connections for the episode in as well
instead of just the manual test one?

**Comment by seertenedos** [comment]
just an update i have worked around the aid lookup for now but it does cause a few hits to your service to work out the uranime ID.  Basically i do a name search and then a details request on each anime till i find the one with the matching AID.  This means that i am now really close to completing the file/series importer with it just needing to do the file post processing but i need the EID for the episodes for that to work.  For now i am just going to use the one series you manually populated for me.  Any idea how long till the rest can be populated?

**Comment by groenlid** [comment]
Should be up and running now after this commit. https://github.com/groenlid/Uranime-api/commit/27829a72aeebefdcf0e17c40facb15721fcc0f50

You can see an example here. http://urani.me:3000/api/anime?source_id=8229&site_id=1

**Comment by seertenedos** [comment]
prefect! thanks

**Comment by groenlid** [comment]
Are you searching through the local information in XDM for the anidb-id before hitting the server? 
Didn't see it in your code when looking through it.

**Comment by seertenedos** [comment]
it is in my media adder code here
https://github.com/seertenedos/xdm-anime-plugin-repo/blob/AniDbFileDetection/AniDbFileBasedMediaAdder/AniDbFileBasedMediaAdder.py

at around line 97.

It is working great and but i just need to fix one thing which is
triggering post processing of the files to move them.  Need to get Lad's
help for that.

Where are we at on the EID for all episodes?  I am currently using episode
number but i would love to switch to EID.

**Comment by groenlid** [comment]
nice :+1: I'll probably port the old import schedule-jobs over to the new api this weekend and add the episode-connection then, but I would not rely on it quite yet. 
It's great if you're fine with using the episode-number (which should match 99.9 percent of the time with anidb) until that's ready.

**Comment by seertenedos** [comment]
Yea it is OK for now as I am still testing. Once I fully test the whole
process I will import my whole collection but that will take days so making
sure everything is working perfectly first

**Comment by carze** [comment]
How would I go about pulling this into my XDM instance for testing purposes? Tried adding @seertenedos repository but it kept on showing up as "unknowing repository"

**Comment by seertenedos** [comment]
I need to update the package config files and I have it in the folder
locally so I did not need the config. I will try to update this weekend and
get back to you. If I don't contact you by Monday remind me.

**Comment by groenlid** [comment]
Either that, or you can clone/fork @seertenedos's repo.
Hmm.. I guess I also could merge @seertenedos and @lad1337's changes into my main branch since most people probably want those features. What's the status @seertenedos? Are your featurebranches stable enough for master?

**Comment by seertenedos** [comment]
Only thing mine is missing is setting series status to wanted after
importing all the episodes you already have. Of you are not using that the
rest is stable. I hope to food the mass import tomorrow
